src/Aspire.Hosting/ApplicationModel/ContainerPullPolicyAnnotation.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Image pull policies for container resources.
+/// </summary>
+public enum ImagePullPolicy
+{
+    /// <summary>
+    /// Default image pull policy behavior. Currently this will be the same as the default behavior for your container runtime.
+    /// </summary>
+    Default = 0,
+    /// <summary>
+    /// Always pull the image when creating the container.
+    /// </summary>
+    Always,
+    /// <summary>
+    /// Pull the image only if it does not already exist.
+    /// </summary>
+    Missing,
+}
+
+/// <summary>
+/// Annotation that controls the image pull policy for a container resource.
+/// </summary>
+[DebuggerDisplay("Type = {GetType().Name,nq}")]
+public sealed class ContainerImagePullPolicyAnnotation : IResourceAnnotation
+{
+    /// <summary>
+    /// Gets or sets the image pull policy for the container resource.
+    /// </summary>
+    public required ImagePullPolicy ImagePullPolicy { get; set; }
```

Comment:
@mitchdenny - do we usually have immutable annotation classes? Should this be a ctor parameter and a get-only property?

Reply:
I based the implementation off ContainerImageAnnotation, which uses this pattern, but there are others that use constructors. I'd argue record types could be a good fit for the annotations if we ever decided like making that much of a breaking change.